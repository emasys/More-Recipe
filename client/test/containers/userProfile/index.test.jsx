import React from 'react';
import { shallow, mount } from 'enzyme';
import Router from 'react-mock-router';

// Component
import {
  UserProfile,
  mapStateToProps
} from '../../../src/containers/UserProfile/index';
import fakeStore from '../../__mocks__/fakeStore';

let wrapper = null;

jest.mock('../../../src/components/CatalogList');
jest.mock('../../../src/components/Navbar');

describe('Test suite for User profile component', () => {
  const props = {
    userInfo: fakeStore.user.userInfo,
    getUserRecipes: jest.fn(),
    clearRecipes: jest.fn(),
    user: [],
    getUserInfo: jest.fn(),
    match: { params: { id: 1 } },
    count: 1,
    recipes: fakeStore.recipes.userRecipes
  };
  beforeEach(() => {
    wrapper = mount(<Router>
      <UserProfile {...props} />
    </Router>);
  });
  it('Should render with props', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('small.header-title').text()).toMatch('admin');
  });

  it('Should render without props', () => {
    const miniProps = {
      userInfo: null,
      getUserRecipes: jest.fn(),
      clearRecipes: jest.fn(),
      user: [],
      getUserInfo: jest.fn(),
      match: { params: { id: 1 } },
      count: 1,
      recipes: fakeStore.recipes.userRecipes
    };
    const app = mount(<Router>
      <UserProfile {...miniProps} />
    </Router>);

    expect(app.length).toEqual(1);
    expect(app).toMatchSnapshot();
    expect(app.find('p#loading').text()).toMatch('loading...');
  });

  it('Should render without full names', () => {
    const miniProps = {
      userInfo: {
        data: {
          id: 1,
          firstName: null,
          lastName: null,
          bio: null,
          email: 'emasysnd@gmail.com',
          country: null,
          avatar: null,
          moniker: 'admin'
        }
      },
      getUserRecipes: jest.fn(),
      clearRecipes: jest.fn(),
      user: [],
      getUserInfo: jest.fn(),
      match: { params: { id: 1 } },
      count: 1,
      recipes: fakeStore.recipes.userRecipes
    };
    const app = mount(<Router>
      <UserProfile {...miniProps} />
    </Router>);

    expect(app.length).toEqual(1);
    expect(app).toMatchSnapshot();
    expect(app.find('h2.mb-10.bolder').text()).toMatch('admin');
  });

  it('Should click on the viewMore button', () => {
    const app = shallow(<UserProfile {...props} />);
    const instance = app.instance();
    const view = jest.spyOn(instance, 'viewMore');

    app
      .find('button.btn.btn-lg.btn-outline-dark.text-center')
      .simulate('click', {
        preventDefault: () => {}
      });
  });

  it('should call componentWillReceiveProps and unmount', () => {
    const app = shallow(<UserProfile {...props} />);
    const instance = app.instance();
    const compWRP = jest.spyOn(instance, 'componentWillReceiveProps');
    const newProps = {
      count: 1
    };
    const nextProps = {
      count: 20
    };


    app.setProps(newProps);
    expect(compWRP).toHaveBeenCalled();
    app.setProps(nextProps);
    expect(compWRP).toHaveBeenCalled();

    app.unmount();
  });

  it('should test mapStateToProps', () => {
    const initialState = {
      recipes: { userRecipes: [], userInfo: {}, userRecipesCount: 0 },
      user: { userInfo: {} }
    };
    expect(mapStateToProps(initialState).count).toEqual(0);
    expect(mapStateToProps(initialState).recipes).toEqual([]);
    expect(mapStateToProps(initialState).userInfo).toEqual({});
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
