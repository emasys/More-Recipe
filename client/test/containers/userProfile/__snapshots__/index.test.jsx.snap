// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for User profile component Should render without full names 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <UserProfile
    clearRecipes={[Function]}
    count={1}
    getUserInfo={[Function]}
    getUserRecipes={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    recipes={Array []}
    user={Array []}
    userInfo={
      Object {
        "data": Object {
          "avatar": null,
          "bio": null,
          "country": null,
          "email": "emasysnd@gmail.com",
          "firstName": null,
          "id": 1,
          "lastName": null,
          "moniker": "admin",
        },
      }
    }
  >
    <div>
      <Connect(Navbar)
        className="bg-dark fixed-top"
      />
      <section
        className="container-fluid profile catalog-wrapper-full-grid mt-70"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-lg-3 col-md-4 col-sm-12 mh-50"
          >
            <UserProfileInfo
              data={
                Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": null,
                    "email": "emasysnd@gmail.com",
                    "firstName": null,
                    "id": 1,
                    "lastName": null,
                    "moniker": "admin",
                  },
                }
              }
            >
              <div
                className="col-lg-3 col-md-4 col-sm-12 mb-10 fixed"
              >
                <img
                  alt="avi"
                  className="img-fluid rounded mb-3"
                  src="https://res.cloudinary.com/emasys/image/upload/v1512284260/mejglndrmulkudy2feug.png"
                />
                <div
                  className="bg-light rounded p-10 profile-wrapper"
                >
                  <h2
                    className="mb-10 bolder"
                  >
                    admin
                  </h2>
                  <p />
                  <hr />
                  <small>
                    <i
                      aria-hidden="true"
                      className="fa fa-envelope"
                    />
                     
                    emasysnd@gmail.com
                  </small>
                </div>
              </div>
            </UserProfileInfo>
          </div>
          <div
            className="col-lg-9 col-md-8 col-sm-12 recipe-lists"
          >
            <div
              className="clearfix"
            >
              <h4
                className="header-title float-left clearfix"
              >
                admin
                '
                s recipes
                 
                <span
                  className="badge badge-dark"
                  data-tip="Total number of recipes added"
                >
                  1
                </span>
              </h4>
            </div>
            <hr />
            <div
              className="row justify-content-center"
            >
              <Connect(CatalogList)
                catalog={Array []}
                clearRecipes={[Function]}
                count={1}
                getUserInfo={[Function]}
                getUserRecipes={[Function]}
                match={
                  Object {
                    "params": Object {
                      "id": 1,
                    },
                  }
                }
                recipes={Array []}
                showDeleteBtn={false}
                user={Array []}
                userInfo={
                  Object {
                    "data": Object {
                      "avatar": null,
                      "bio": null,
                      "country": null,
                      "email": "emasysnd@gmail.com",
                      "firstName": null,
                      "id": 1,
                      "lastName": null,
                      "moniker": "admin",
                    },
                  }
                }
              />
            </div>
            <div
              className="row justify-content-center"
            >
              <button
                className=" btn btn-lg btn-outline-dark text-center"
                onClick={[Function]}
              >
                View more
              </button>
            </div>
          </div>
        </div>
        <ReactTooltip
          effect="float"
          insecure={true}
          place="bottom"
          resizeHide={true}
          type="dark"
          wrapper="div"
        >
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          />
        </ReactTooltip>
      </section>
    </div>
  </UserProfile>
</MockRouter>
`;

exports[`Test suite for User profile component Should render without props 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <UserProfile
    clearRecipes={[Function]}
    count={1}
    getUserInfo={[Function]}
    getUserRecipes={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    recipes={Array []}
    user={Array []}
    userInfo={null}
  >
    <div>
      <Connect(Navbar)
        className="bg-dark fixed-top"
      />
      <section
        className="container-fluid profile catalog-wrapper-full-grid mt-70"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-lg-3 col-md-4 col-sm-12 mh-50"
          >
            <UserProfileInfo
              data={null}
            >
              <p
                id="loading"
              >
                loading...
              </p>
            </UserProfileInfo>
          </div>
          <div
            className="col-lg-9 col-md-8 col-sm-12 recipe-lists"
          >
            <div
              className="clearfix"
            >
              <h4
                className="header-title float-left clearfix"
              >
                loading
                '
                s recipes
                 
                <span
                  className="badge badge-dark"
                  data-tip="Total number of recipes added"
                >
                  1
                </span>
              </h4>
            </div>
            <hr />
            <div
              className="row justify-content-center"
            >
              <Connect(CatalogList)
                catalog={Array []}
                clearRecipes={[Function]}
                count={1}
                getUserInfo={[Function]}
                getUserRecipes={[Function]}
                match={
                  Object {
                    "params": Object {
                      "id": 1,
                    },
                  }
                }
                recipes={Array []}
                showDeleteBtn={false}
                user={Array []}
                userInfo={null}
              />
            </div>
            <div
              className="row justify-content-center"
            >
              <button
                className=" btn btn-lg btn-outline-dark text-center"
                onClick={[Function]}
              >
                View more
              </button>
            </div>
          </div>
        </div>
        <ReactTooltip
          effect="float"
          insecure={true}
          place="bottom"
          resizeHide={true}
          type="dark"
          wrapper="div"
        >
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          />
        </ReactTooltip>
      </section>
    </div>
  </UserProfile>
</MockRouter>
`;

exports[`Test suite for User profile component should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <UserProfile
    clearRecipes={[Function]}
    count={1}
    getUserInfo={[Function]}
    getUserRecipes={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    recipes={Array []}
    user={Array []}
    userInfo={
      Object {
        "data": Object {
          "avatar": null,
          "bio": null,
          "country": "Nigeria",
          "email": "emasysnd@gmail.com",
          "firstName": "emmy",
          "id": 1,
          "lastName": "endy",
          "moniker": "admin",
        },
        "success": true,
      }
    }
  >
    <div>
      <Connect(Navbar)
        className="bg-dark fixed-top"
      />
      <section
        className="container-fluid profile catalog-wrapper-full-grid mt-70"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-lg-3 col-md-4 col-sm-12 mh-50"
          >
            <UserProfileInfo
              data={
                Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": "Nigeria",
                    "email": "emasysnd@gmail.com",
                    "firstName": "emmy",
                    "id": 1,
                    "lastName": "endy",
                    "moniker": "admin",
                  },
                  "success": true,
                }
              }
            >
              <div
                className="col-lg-3 col-md-4 col-sm-12 mb-10 fixed"
              >
                <img
                  alt="avi"
                  className="img-fluid rounded mb-3"
                  src="https://res.cloudinary.com/emasys/image/upload/v1512284260/mejglndrmulkudy2feug.png"
                />
                <div
                  className="bg-light rounded p-10 profile-wrapper"
                >
                  <h2
                    className="mb-10 bolder"
                  >
                    emmy endy 
                    (
                    <small
                      className="header-title"
                    >
                      admin
                    </small>
                    )
                  </h2>
                  <p />
                  <hr />
                  <small>
                    <i
                      aria-hidden="true"
                      className="fa fa-envelope"
                    />
                     
                    emasysnd@gmail.com
                  </small>
                  <p
                    className=" text-capitalize"
                  >
                    <small>
                      <i
                        aria-hidden="true"
                        className="fa fa-map-marker"
                      />
                       
                      Nigeria
                    </small>
                  </p>
                </div>
              </div>
            </UserProfileInfo>
          </div>
          <div
            className="col-lg-9 col-md-8 col-sm-12 recipe-lists"
          >
            <div
              className="clearfix"
            >
              <h4
                className="header-title float-left clearfix"
              >
                admin
                '
                s recipes
                 
                <span
                  className="badge badge-dark"
                  data-tip="Total number of recipes added"
                >
                  1
                </span>
              </h4>
            </div>
            <hr />
            <div
              className="row justify-content-center"
            >
              <Connect(CatalogList)
                catalog={Array []}
                clearRecipes={[Function]}
                count={1}
                getUserInfo={[Function]}
                getUserRecipes={[Function]}
                match={
                  Object {
                    "params": Object {
                      "id": 1,
                    },
                  }
                }
                recipes={Array []}
                showDeleteBtn={false}
                user={Array []}
                userInfo={
                  Object {
                    "data": Object {
                      "avatar": null,
                      "bio": null,
                      "country": "Nigeria",
                      "email": "emasysnd@gmail.com",
                      "firstName": "emmy",
                      "id": 1,
                      "lastName": "endy",
                      "moniker": "admin",
                    },
                    "success": true,
                  }
                }
              />
            </div>
            <div
              className="row justify-content-center"
            >
              <button
                className=" btn btn-lg btn-outline-dark text-center"
                onClick={[Function]}
              >
                View more
              </button>
            </div>
          </div>
        </div>
        <ReactTooltip
          effect="float"
          insecure={true}
          place="bottom"
          resizeHide={true}
          type="dark"
          wrapper="div"
        >
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          />
        </ReactTooltip>
      </section>
    </div>
  </UserProfile>
</MockRouter>
`;
